app/src/main/kotlin/edu/stanford/bdh/engagehf/navigation/screens/AppScreenViewModel.kt

```diff
@@ -74,9 +93,34 @@ class AppScreenViewModel @Inject constructor(
             is Action.UpdateSelectedBottomBarItem -> {
                 _uiState.update { it.copy(selectedItem = action.selectedBottomBarItem) }
             }
+
             is Action.UpdateBottomSheetState -> {
                 _uiState.update { it.copy(isBottomSheetExpanded = action.isExpanded) }
             }
+
+            is Action.ShowDialog -> {
+                _uiState.update { it.copy(appTopBar = it.appTopBar.copy(showDialog = action.showDialog)) }
+            }
+
+            Action.ShowHealthSummary -> {
+                viewModelScope.launch {
+                    _uiState.update { it.copy(appTopBar = it.appTopBar.copy(isHealthSummaryLoading = true)) }
+                    healthSummaryService.generateHealthSummaryPdf()
+                    _uiState.update {
+                        it.copy(
+                            appTopBar = it.appTopBar.copy(
+                                isHealthSummaryLoading = false,
+                                showDialog = false
+                            )
+                        )
+                    }
+                }
+            }
+
+            Action.SignOut -> {
+                _uiState.update { it.copy(appTopBar = it.appTopBar.copy(showDialog = false)) }
```

Comment:
Let's create an issue for this. It does not only require signing out from firebase, but also clearing the navigation back stack and popping to the initial onboarding screen (also setting `MainUiState.Content(Routes.OnboardingScreen)` in `MainActivityViewModel`

Reply:
#89